src/sdk/pynni/nni/compression/torch/builtin_quantizers.py

```diff
@@ -29,23 +29,182 @@ class QAT_Quantizer(Quantizer):
     """
     def __init__(self, model, config_list):
         """
-        config_list: supported keys:
-            - q_bits
+        Create a wrapper forward function to replace the original one.
+
+        Parameters
+        ----------
+        layer : LayerInfo
+            the layer to quantize
+        config_list : list of dict
+            list of configurations for quantization
+            supported keys for dict:
+                - quant_types : list of string
+                    type of quantization you want to apply, currently support 'weight', 'input', 'output'
+                - quant_bits : dict
+                    bits length of quantization, key is the quantization type, value is the length, eg. {'weight', 8}, default value is 8
+                - quant_start_step : int
+                    disable quantization until model are run by certain number of steps, this allows the network to enter a more stable
+                    state where activation quantization ranges do not exclude a signiﬁcant fraction of values, default value is 0
+                - op_types : list of string
+                    types of nn.module you want to apply quantization, eg. 'Conv2d'
         """
         super().__init__(model, config_list)
+        self.modules_to_compress = self.detect_modules_to_compress()
```

Comment:
Do not assign to same attribute in both base and derived classes.

Reply:
change to `modules_to_compress = self.detect_modules_to_compress()`